### PR TITLE
[GOCOMICS-1070] default to helm whitelist values

### PR DESCRIPTION
**Pull request by ebronson68**
Added additional checks to default to the whitelist values set in the Helm values.yaml file.

Related build: [Test AKS Deploy](https://github.com/Andrews-McMeel-Universal/reusable_workflows-test/actions/workflows/test-aks-deploy.yaml)

### CODE REVIEW

File: .github/workflows/aks-deploy.yaml [hunk 1]
```diff
@@ -301,48 +307,52 @@ jobs:
...
               nginx.ingress.kubernetes.io/auth-type: "basic"
               nginx.ingress.kubernetes.io/auth-secret: "${{ env.appName }}-basic-auth"
               nginx.ingress.kubernetes.io/auth-realm: "Authentication Required"
           adminingress:
             annotations:
-              nginx.ingress.kubernetes.io/whitelist-source-range: "${{ inputs.adminIngressWhitelist }}"
+              nginx.ingress.kubernetes.io/whitelist-source-range: "${ADMIN_INGRESS_WHITELIST}"
           EOF
-          elif [[ ${ADMIN_INGRESS_WHITELIST} != "207.67.20.252" ]] || [[ ${INGRESS_WHITELIST} != "0.0.0.0/0" ]]; then
+          elif [[ "${ADMIN_INGRESS_WHITELIST}" != "207.67.20.252" ]] || [[ "${INGRESS_WHITELIST}" != "0.0.0.0/0" ]]; then
```
Comment (abarrows) [inline .github/workflows/aks-deploy.yaml:340]: Should we put this in an org level secret?

File: .github/workflows/aks-deploy.yaml [hunk 2]
```diff
@@ -186,6 +187,9 @@ jobs:
             }
           }
           
+          $release = "$appName-${{ github.sha }}".Substring(0,53)
```
Comment (abarrows) [inline .github/workflows/aks-deploy.yaml:190]: On the Substring call here is there a more reliable way to assign this?  I don't know what I don't know here but if that $release var gets viewed in K8s or docker, I wonder if we could make it something like:
$appName-$appEnvironment-$appVersion-(last 10 characters of the github sha) or something like that, maybe?  Not a show stopper just thought I'd ask!

File: .github/workflows/aks-deploy.yaml [hunk 3]
```diff
@@ -301,48 +307,52 @@ jobs:
...
-          release=$(echo ${release//[!0-9a-zA-Z]/-} | tr '[:upper:]' '[:lower:]' | sed -e 's/^-/z/' -e 's/-$/z/')
-          echo "release=$release" >> $GITHUB_ENV
-
       - name: Create values override file
         run: |
           WEB_AUTHENTICATION="${{ inputs.webAuthentication }}"
-          ADMIN_INGRESS_WHITELIST="${{ inputs.adminIngressWhitelist }}"
-          INGRESS_WHITELIST="${{ inputs.ingressWhitelist }}"
-          if [[ ${WEB_AUTHENTICATION} == "true" ]] ; then
+          INGRESS_WHITELIST_ENV="${{ env.ingressWhitelist }}"
```
Comment (abarrows) [inline .github/workflows/aks-deploy.yaml:313]: What type of info is in the INGRESS_WHITELIST?  Is this like IP addresses internally?

File: .github/workflows/workflow-linter.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+name: Lint GitHub Actions workflows
```
Comment (abarrows) [inline .github/workflows/workflow-linter.yaml:1]: 🏅

**Review by abarrows — APPROVED** [review]
See the comments below but it looks good from what I can see!  If we merge this in, will there be any risk to the products using it or is it protected under a @version number?

File: .github/workflows/aks-deploy.yaml (hunk 1 above)
Comment (ebronson68) [reply .github/workflows/aks-deploy.yaml:340]: Putting them in a org level secret might cause more confusion than what it is worth. The default values in the inputs already account for an org-level standard. 

Though, checking whether the default value is being used or not in that if/else condition could be improved most likely.

File: .github/workflows/aks-deploy.yaml (hunk 2 above)
Comment (ebronson68) [reply .github/workflows/aks-deploy.yaml:190]: Well previously, this was set by a whole other step entirely. I just moved it to this step to fix a silly linting error I was getting.

Here is where it used to be before: https://github.com/Andrews-McMeel-Universal/reusable_workflows/pull/10/files#diff-377d534d3f697f70d4e4ae1913ade4d3d1ef69befae238c4af735a7a0a5558faL304-L311

As far as changing it to simplify the image tag or doing something similar goes, it wouldn't make too much a difference if we name it differently as we really don't touch those images once they've been added to the registry unless I'm forgetting about something.

Maybe once we figure out a good way to use existing images when promoting to the different environments, we could revisit this though.

File: .github/workflows/aks-deploy.yaml (hunk 3 above)
Comment (ebronson68) [reply .github/workflows/aks-deploy.yaml:313]: This new variable is to allow for transitioning away from and consolidating the multiple Helm environment directories in repositories. This is one of the few Helm values that are set on a per-environment basis. 

Setting it in the reusable workflow/inputs here allows us to change the "whitelist-source-range" value for the K8s ingress for the application. 

The "whitelist-source-range" annotation just acts as a simple ACL to explicitly allow traffic from specific IPs. For example, we set it to "0.0.0.0/0" for production as we don't want to restrict traffic to the application from the Internet. For development, we usually lock it down to just the Boley IP.

**Comment by ebronson68** [comment]
> See the comments below but it looks good from what I can see! If we merge this in, will there be any risk to the products using it or is it protected under a @Version number?

So we are actually using it in a few environments already (we currently have releases 2.1.11 and 2.1.12 pointing to a slightly older version of this branch before I fixed the linting errors). I've already tested it in the new reusable_workflows-test repository plenty of times and have used the new linter on it enough to know there aren't any issues with the workflows. 

We just haven't merged in this changes yet.

**Pull request merged**